### ISSUE TITLE
be the tensorflow param 'dynamic_pad' support by keras ?

### ISSUE THREAD

**Issue by xylcbd** [fchollet/keras#3999]
My question is like this below : 

> Tensorflow’s RNN functions expect a tensor of shape [B, T, ...] as input, where B is the batch size and T is the length in time of each input (e.g. the number of words in a sentence). The last dimensions depend on your data. Do you see a problem with this? Typically, not all sequences in a single batch are of the same length T, but in order to feed them into the RNN they must be. Usually that’s done by padding them: Appending 0‘s to examples to make them equal in length.
> 
> Now imagine that one of your sequences is of length 1000, but the average length is 20. If you pad all your examples to length 1000 that would be a huge waste of space (and computation time)! That’s where batch padding comes in. If you create batches of size 32, you only need to pad examples within the batch to the same length (the maximum length of examples in that batch). That way, a really long example will only affect a single batch, not all of your data.
> 
> That all sounds pretty messy to deal with. Luckily, Tensorflow has built-in support for batch padding. If you set dynamic_pad=True when calling tf.train.batch the returned batch will be automatically padded with 0s. Handy! A lower-level option is to use tf.PaddingFIFOQueue.

From : http://www.wildml.com/2016/08/rnns-in-tensorflow-a-practical-guide-and-undocumented-features/

and , so , be the tensorflow param 'dynamic_pad' support by keras ?

**Comment by xylcbd** [comment]
or , is there any method to resolve this issue ? thx.

**Comment by kgrm** [comment]
Keras does not support this tensorflow feature, but you should use the Masking layer to achieve the same. It works by making the rnn skip masked timesteps, allowing training on arbitrary-length sequences.

**Comment by xylcbd** [comment]
ok , and thanks.
